dproc/dproc.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package dproc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	dataproc "cloud.google.com/go/dataproc/apiv1"
+	"cloud.google.com/go/dataproc/apiv1/dataprocpb"
+	"github.com/cloudspannerecosystem/harbourbridge/common/constants"
+	"github.com/cloudspannerecosystem/harbourbridge/profiles"
+	"google.golang.org/api/option"
+)
+
+const (
+	runtimeVersion string = "1.1"
+	mainClass      string = "com.google.cloud.dataproc.templates.main.DataProcTemplate"
+	fetchSize      string = "500"
+	batchSize      string = "500"
+	saveMode       string = "Append"
+)
+
+type DataprocRequestParams struct {
+	Project       string
+	TemplateArgs  []string
+	JarFileUris   []string
+	SubnetworkUri string
+	Location      string
+}
+
+func GetDataprocRequestParams(sourceProfile profiles.SourceProfile, targetProfile profiles.TargetProfile, srcSchema string, srcTable string, primaryKeys string, location string, subnet string) (DataprocRequestParams, error) {
```

Comment:
Please add unit tests

Reply:
Added some base unit tests for method `TestGetDataprocRequestParams()`, there's still scope for more test cases for same and other methods, for now added TODO comments for same, will add them eventually
Commit: https://github.com/cloudspannerecosystem/harbourbridge/pull/520/commits/e85e5b1db50678234e466fd266f5484e7aaae492